File: esm-android/app/src/main/java/com/example/aed_drone/Analyze_Web_Command.java
package com.example.aed_drone;

import android.os.Handler;
import android.text.format.DateFormat;
import android.util.Log;

import org.json.JSONObject;

import java.util.Calendar;

public class Analyze_Web_Command {

    private static final Handler mHandler = MainActivity.mHandler;

    public static void Analyze_JSON(String msg_from_web){
        JSONObject json_msg;
        try {
            json_msg = new JSONObject(msg_from_web);
            Object jsonOb = json_msg.get("cmd");
            String web_command = String.valueOf(jsonOb);

            Calendar mCal = Calendar.getInstance();
            CharSequence date_str  = DateFormat.format("yyyy-MM-dd kk:mm:ss", mCal.getTime());
            mHandler.obtainMessage(109, date_str + ": " + web_command).sendToTarget();

            switch (web_command) {
                case "TAKEOFF":
                    Object takeoff_height = json_msg.get("altitude");
                    float height = ObjectToFloat(takeoff_height);
                    Drone_Command.TAKEOFF_FUNCTION(height);
                    break;
                case "ARM":
                    Drone_Command.ARM_DISARM_FUNCTION(1);
                    break;
                case "DISARM":
                    Drone_Command.ARM_DISARM_FUNCTION(0);
                    break;
                case "GOTO":
                    Object goTo_height = json_msg.get("altitude");
                    Object goTo_lat = json_msg.get("lat");
                    Object goTo_lng = json_msg.get("lng");
                    float altitude = ObjectToFloat(goTo_height);
                    double lat = ObjectToFloat(goTo_lat);
                    double lng = ObjectToFloat(goTo_lng);
                    Drone_Command.GOTO_FUNCTION(lat, lng, altitude);
                    break;
                case "CHANGE_SPEED":
                    Object change_speed = json_msg.get("speed");
                    float speed = ObjectToFloat(change_speed);
                    Drone_Command.CHANGE_SPEED_FUNCTION(speed);
                    break;
                case "CHANGE_YAW":
                    Object change_yaw = json_msg.get("angle");
                    float angle = ObjectToFloat(change_yaw);
                    Drone_Command.CHANGE_YAW_FUNCTION(angle);
                    break;
                // SET_MODE
                case "GUIDED":
                    Drone_Command.SET_MODE_FUNCTION(4);
                    break;
                case "RTL":
                    Drone_Command.SET_MODE_FUNCTION(6);
                    break;
                case "LAND":
                    Drone_Command.SET_MODE_FUNCTION(9);
                    break;
                // SERVO_CONTROL
                case "SERVO_UP":
                    Drone_Command.SERVO_FUNCTION(9, 600);
                    break;
                case "SERVO_DOWN":
                    Drone_Command.SERVO_FUNCTION(9, 2400);
                    break;
                case "SERVO_STOP":
                    Drone_Command.SERVO_FUNCTION(9, 1500);
                    break;
                // GIMBAL
                case "GIMBAL_FRONT_BACK":
                    Object GIMBAL_FRONT_BACK_range = json_msg.get("pwm");
                    int range_FB = ObjectToInt(GIMBAL_FRONT_BACK_range);
                    Drone_Command.SERVO_FUNCTION(10, range_FB);
                    break;
                case "GIMBAL_LEFT_RIGHT":
                    Object GIMBAL_LEFT_RIGHT_range = json_msg.get("pwm");
                    int range_LF = ObjectToInt(GIMBAL_LEFT_RIGHT_range);
                    Drone_Command.SERVO_FUNCTION(11, range_LF);
                    break;
                default:
                    Log.i("Analyze_Web_Command", "Exception WEB Command: " + web_command);
                    break;
            }
        }catch(Exception e){
            e.printStackTrace();
        }
    }

    public static int ObjectToInt(Object json_obj){
        String s = String.valueOf(json_obj);
        return Integer.parseInt(s);
    }

    public static float ObjectToFloat(Object json_obj){
        String s = String.valueOf(json_obj);
        return Float.parseFloat(s);
    }
}

File: vr_sim/virtual-drone-connection/AED-Rescue-Drone-Project-for-Python-main/analyze_web_command.py
from __future__ import print_function
import json
from drone_command import Drone_command

def analyze_web_json(vehicle, msg_from_web):

    msg_from_web_ = msg_from_web.decode("utf-8") 
    msg_from_web_dict = json.loads(msg_from_web_)

    if msg_from_web_dict['cmd'] == "TAKEOFF":
        takeoff_height = msg_from_web_dict['altitude']
        Drone_command.SET_MODE(vehicle, 4)
        Drone_command.ARM_DISARM(vehicle, 1)
        Drone_command.TAKEOFF(vehicle, float(takeoff_height))

    elif msg_from_web_dict['cmd'] == "ARM":
        Drone_command.ARM_DISARM(vehicle, 1)

    elif msg_from_web_dict['cmd'] == "DISARM":
        Drone_command.ARM_DISARM(vehicle, 0)

    elif msg_from_web_dict['cmd'] == "GOTO":
        goTo_height = float(msg_from_web_dict['altitude'])
        goTo_lat= float(msg_from_web_dict['lat'])
        goTo_lng = float(msg_from_web_dict['lng'])
        Drone_command.GOTO(vehicle, goTo_lat, goTo_lng, goTo_height)

    elif msg_from_web_dict['cmd'] == "CHANGE_SPEED":
        change_speed = float(msg_from_web_dict['speed'])
        Drone_command.CHANGE_SPEED(vehicle, change_speed)

    elif msg_from_web_dict['cmd'] == "CHANGE_YAW":
        change_angle = float(msg_from_web_dict['angle'])
        Drone_command.CHANGE_YAW(vehicle, change_angle)

    elif msg_from_web_dict['cmd'] == "GUIDED":
        Drone_command.SET_MODE(vehicle, 4)

    elif msg_from_web_dict['cmd'] == "RTL":
        Drone_command.SET_MODE(vehicle, 6)

    elif msg_from_web_dict['cmd'] == "LAND":
        Drone_command.SET_MODE(vehicle, 9)

    elif msg_from_web_dict['cmd'] == "SERVO_UP":
        Drone_command.SERVO(vehicle, 9, 600)

    elif msg_from_web_dict['cmd'] == "SERVO_DOWN":
        Drone_command.SERVO(vehicle, 9, 2400)

    elif msg_from_web_dict['cmd'] == "SERVO_STOP":
        Drone_command.SERVO(vehicle, 9, 1500)

    elif msg_from_web_dict['cmd'] == "GIMBAL_FRONT_BACK":
        range_ = int(msg_from_web_dict['pwm'])
        Drone_command.SERVO(vehicle, 10, range_)

    elif msg_from_web_dict['cmd'] == "GIMBAL_LEFT_RIGHT":
        range_ = int(msg_from_web_dict['pwm'])
        Drone_command.SERVO(vehicle, 11, range_)

    else:
        print('Analyze_web_command Error...')
